R/timeseriessummary-data.R

```diff
@@ -524,6 +525,178 @@ parseTSSGapTolerances <- function(reportData, timezone){
+#' @param timezone the timezone to parse data into
+#' @return corrections formatted better for report display
+#' @importFrom dplyr rowwise mutate ungroup
+unNestCorrectionParameters <- function(corrections, timezone) {
+  
+  type <- ".dplyr"
+  Offset <- ".dplyr"
+  DriftPoints <- ".dplyr"
+  Value <- ".dplyr"
+  StartShiftPoints <- ".dplyr"
+  EndShiftPoints <- ".dplyr"
+  UsgsType <- ".dplyr"
+  UpperThresholdPoints <- ".dplyr"
+  LowerThresholdPoints <- ".dplyr"
+  ResamplePeriod <- ".dplyr"
+  GapLimit <- ".dplyr"
+  DeviationValue <- ".dplyr"
+  DeviationType <- ".dplyr"
+  WindowSizeInMinutes <- ".dplyr"
+  ungroup <- ".dplyr"
+  
+  params <- corrections$parameters
+  corrections$parameters <- NULL
+  
+  if (!isEmptyVar(params)) {
+    corrections <- cbind(corrections, params)
+  }
+  
+  corrections_formatted <- corrections %>%
+    rowwise() %>%
+    mutate(timezone=timezone) %>%
+    mutate(
+      formattedParameters = switch(type,
+                                   "Offset" = formatCorrectionsParamOffset(Offset),
+                                   "Drift" = formatCorrectionsParamDrift(DriftPoints, timezone),
+                                   "SinglePoint" = formatCorrectionsParamSinglePoint(Value),
+                                   "USGSMultiPoint" = formatCorrectionsParamUSGSMultiPoint(StartShiftPoints, EndShiftPoints, UsgsType),
+                                   "AdjustableTrim" = formatCorrectionsParamAdjustableTrim(UpperThresholdPoints, LowerThresholdPoints, timezone),
+                                   "FillGaps" = formatCorrectionsParamFillGaps(ResamplePeriod, GapLimit),
+                                   "Deviation" = formatCorrectionsParamDeviation(DeviationValue, DeviationType, WindowSizeInMinutes),
+                                   type)) %>% 
+    ungroup()
+  
+  
+  return(corrections_formatted)
+}
+
+#' formats the offset correction parameters
+#' @description formats the offset correction parameters
+#' @param offset the offset value for the parameter offset
+#' @return formatted string of offset parameters for report display
+formatCorrectionsParamOffset <- function(offset) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(offset)) {
+    formattedParameters <- paste0("Offset ", offset)
+  }
+  return(formattedParameters)
+}
+
+#' formats the drift correction parameters
+#' @description formats the drift correction parameters
+#' @param driftPoints the driftPoints values in a list including
+#' date/times and the difference values in feet
+#' @param timezone the timezone for the drift parameters
+#' @return formatted string of drift parameters for report display
+formatCorrectionsParamDrift <- function(driftPoints, timezone) {
+  formattedParameters <- ""
+  if (!isEmptyOrBlank(driftPoints) && !isEmptyOrBlank(timezone)) {
+    driftPoints <- as.data.frame(driftPoints)
+    if (any(names(driftPoints) %in% c("Value","Offset"))) {
+        for (i in 1:nrow(driftPoints)) {
```

Comment:
What if "value" was in the data, but "Offset" wasn't? This would fail since you are saying either to get into the if code, but only "offset" is used